### PR TITLE
refactor: DiscoverSriovDevices imporvements

### PR DESCRIPTION
**Pull request by adrianchiris**
- Move check for supported model earlier in the loop to exit early  in case of unsupported device

- Use Netlink Link object to get some PF information instead of sysfs

- Use Netlink Link object to get some VF information instead of sysfs

### CODE REVIEW

File: pkg/host/internal/sriov/sriov.go [hunk 1]
```diff
@@ -115,12 +115,15 @@ func (s *sriov) GetVfInfo(pciAddr string, devices []*ghw.PCIDevice) sriovnetwork
...
 	if name := s.networkHelper.TryGetInterfaceName(pciAddr); name != "" {
-		vf.Name = name
-		vf.Mac = s.networkHelper.GetNetDevMac(name)
+		link, err := s.netlinkLib.LinkByName(name)
+		if err != nil {
+			log.Log.Error(err, "getVfInfo(): unable to get VF Link Object", "name", name, "device", pciAddr)
+		} else {
+			vf.Name = name
+			vf.Mtu = link.Attrs().MTU
+			vf.Mac = string(link.Attrs().HardwareAddr)
```
Comment (ykulazhenkov) [inline pkg/host/internal/sriov/sriov.go:125]: This construction will not work. You should use `link.Attrs().HardwareAddr.String()`

File: pkg/host/internal/sriov/sriov.go [hunk 2]
```diff
@@ -115,12 +115,15 @@ func (s *sriov) GetVfInfo(pciAddr string, devices []*ghw.PCIDevice) sriovnetwork
...
-		vf.Mtu = mtu
-	}
 	if name := s.networkHelper.TryGetInterfaceName(pciAddr); name != "" {
-		vf.Name = name
-		vf.Mac = s.networkHelper.GetNetDevMac(name)
+		link, err := s.netlinkLib.LinkByName(name)
+		if err != nil {
+			log.Log.Error(err, "getVfInfo(): unable to get VF Link Object", "name", name, "device", pciAddr)
+		} else {
+			vf.Name = name
```
Comment (ykulazhenkov) [inline pkg/host/internal/sriov/sriov.go:123]: Do we want to keep Name set in case of failure?

File: pkg/host/internal/sriov/sriov.go [hunk 3]
```diff
@@ -221,44 +223,41 @@ func (s *sriov) DiscoverSriovDevices(storeManager store.ManagerInterface) ([]sri
...
-		if err != nil {
-			log.Log.Error(err, "DiscoverSriovDevices(): unable to get device names for device, skipping", "device", device)
-			continue
-		}
+		pfNetName := s.networkHelper.TryGetInterfaceName(device.Address)
 
-		if len(deviceNames) == 0 {
-			// no network devices found, skipping device
+		if pfNetName == "" {
+			log.Log.Error(err, "DiscoverSriovDevices(): unable to get device names for device, skipping", "device", device.Address)
```
Comment (ykulazhenkov) [inline pkg/host/internal/sriov/sriov.go:242]: nit: names > name

File: pkg/host/internal/sriov/sriov.go [hunk 4]
```diff
@@ -221,44 +223,41 @@ func (s *sriov) DiscoverSriovDevices(storeManager store.ManagerInterface) ([]sri
...
-		if mtu := s.networkHelper.GetNetdevMTU(device.Address); mtu > 0 {
-			iface.Mtu = mtu
-		}
-		if name := s.networkHelper.TryGetInterfaceName(device.Address); name != "" {
-			iface.Name = name
-			iface.Mac = s.networkHelper.GetNetDevMac(name)
-			iface.LinkSpeed = s.networkHelper.GetNetDevLinkSpeed(name)
-			iface.LinkType = s.GetLinkType(name)
+			Mtu:        link.Attrs().MTU,
+			Mac:        string(link.Attrs().HardwareAddr),
```
Comment (ykulazhenkov) [inline pkg/host/internal/sriov/sriov.go:258]: use `link.Attrs().HardwareAddr,String()` instead

**Review by ykulazhenkov — COMMENTED** [review]
added few comments

File: pkg/host/internal/sriov/sriov.go (hunk 4 above)
Comment (adrianchiris) [reply pkg/host/internal/sriov/sriov.go:258]: IDE autocompleted it :(. yes i will fix. which probably related to CI failure.

File: pkg/host/internal/sriov/sriov.go (hunk 1 above)
Comment (adrianchiris) [reply pkg/host/internal/sriov/sriov.go:125]: IDE autocompleted it :(. yes i will fix. which probably related to CI failure.

File: pkg/host/internal/sriov/sriov.go (hunk 2 above)
Comment (adrianchiris) [reply pkg/host/internal/sriov/sriov.go:123]: it keeps same logic as before, if TryGetInterfaceName returns an interface, the name is set.

im fine with making this more strict and set if we can get the link device.

File: pkg/host/internal/sriov/sriov.go [hunk 5]
```diff
@@ -221,44 +223,41 @@ func (s *sriov) DiscoverSriovDevices(storeManager store.ManagerInterface) ([]sri
...
-		}
-		if mtu := s.networkHelper.GetNetdevMTU(device.Address); mtu > 0 {
-			iface.Mtu = mtu
-		}
-		if name := s.networkHelper.TryGetInterfaceName(device.Address); name != "" {
-			iface.Name = name
-			iface.Mac = s.networkHelper.GetNetDevMac(name)
-			iface.LinkSpeed = s.networkHelper.GetNetDevLinkSpeed(name)
-			iface.LinkType = s.GetLinkType(name)
+			Mtu:        link.Attrs().MTU,
```
Comment (ykulazhenkov) [inline pkg/host/internal/sriov/sriov.go:257]: you lost the name field here

Comment (adrianchiris) [reply pkg/host/internal/sriov/sriov.go:257]: good catch !

**Review by ykulazhenkov — APPROVED** [review]
LGTM

**Review by SchSeba — APPROVED** [review]
LGTM

nice work!

**Pull request merged**